src/api/routes/thirdparty.js

```
@@ -1008,4 +1008,139 @@ thirdpartyRoutes.post("/mercury/refresh", async (c, next) => {
+    // Gmail returns base64url-encoded RFC 2822 message (no padding per RFC 4648 §5)
+    // Convert base64url → base64 (+ pad) → binary
+    const b64 = data.raw.replace(/-/g, "+").replace(/_/g, "/");
+    const padded = b64 + "=".repeat((4 - (b64.length % 4)) % 4);
+    const binaryString = atob(padded);
+    const bytes = new Uint8Array(binaryString.length);
+    for (let i = 0; i < binaryString.length; i++) {
+      bytes[i] = binaryString.charCodeAt(i);
+    }
+
+    // Strip non-filename-safe chars from messageId to prevent header injection
+    const safeFilename = messageId.replace(/[^a-zA-Z0-9_\-]/g, "_");
```

ESLint

[error] Unnecessary escape character: \-.

(no-useless-escape)